src/pyodide/internal/workers.py

```diff
@@ -604,3 +605,24 @@ async def json(self, **kwargs: Any) -> Any:
     async def text(self) -> str:
         self._raise_if_failed()
         return await self.js_object.text()
+
+
+def event(func):
```

Comment:
I think `handler` would be a better name? What does our documentation call these things?